tools/analysis/programs.go

```diff
@@ -69,22 +69,53 @@ func (programs Programs) load(
 
 	program, err := parser.ParseProgram(nil, code, parser.Config{})
 	if err != nil {
-		return wrapError(err)
+		// If a parser error handler is set and the error is non-fatal
+		// use handler to handle the error (e.g. to report it and continue analysis)
+		// This will only be used for the entry point program (e.g not an imported program)
+		wrappedErr := wrapError(err)
+		if loadError == nil {
+			loadError = wrappedErr
+		}
+
+		if config.HandleParserError != nil && program != nil && importingLocation == nil {
```

Comment:
Why is the handler only called when a program exists and when the parsed program was not imported?

It might be surprising for a user of the package that the handler is not always called.

Maybe pass the details (program, importing location) to the handler, so it can determine / filter on its own, if needed. Or at least leave a comment similar to the HandleCheckerError call below.